src/Components/Atbash/AtbashInput.js

```diff
@@ -19,45 +42,115 @@ class Atbash extends Component {
+    //original = original.substring(0, index) + atbashEncode(original[index]) + original.substring(index + 1);
+    let newmsgs = Array.from(this.state.messages)
+    newmsgs.splice(i, 1, original)
+    console.log(newmsgs[i])
+    this.setState({
+      messages: newmsgs
+    });
+  }
+
+  decrypt = (i) => {
+    let original = this.state.messages[i]
+    for (let index = 0; index !== original.length; index++) {
+      this.timeouts.push(setTimeout(this.decryptHelper(i, index), index * 900))
+    }
+  }
+
+  Ez = (i) => {
+    return <div className="is-size-4 my-5 is-inline-block" id={i} onClick={() => this.decrypt(i)}>{this.state.messages[i]}</div>
+  }
+
+  animateMsg = (message, final=false) => {
+    console.log(message)
+    return(
+      <div className="is-size-4 overlay is-inline-flex">
+          <Anime 
+              rotateY={final ? [-90, 0] : [0, 45]} 
+              opacity={final ? [0, 1] : [1,0]} 
+              duration={final ? "400" : "200" }
+              delay={anime.stagger(50)}
+              easing="easeOutSine">
+                  {this.putCharsInSpans(message)}
+          </Anime>
+      </div>
+    )
+  }
+
+  animateAt = (index) => {
+    let newAnimateState = Array.from(this.state.animateDecrypt)
+    newAnimateState.splice(index, 1, true)
+    this.setState({
+      animateDecrypt: newAnimateState
+    })
+  }
+
+  decryptAt = (index) => {
+    let newAnimateState = Array.from(this.state.animateDecrypt)
+    newAnimateState.splice(index, 1, true)
+    this.setState({
+      decrypted: newAnimateState
+    })
+  }
+
   render() {
+
+    // [pls forgive my spaghetti code]
+    let finalCiphertext = ["", "", ""];
+    let finalPlaintext = ["", "", ""];
+    this.state.messages.forEach((el, i) => {
+      if (this.state.animateDecrypt[i]) {
+        finalCiphertext.splice(i, 1, this.animateMsg(el))
+        finalPlaintext.splice(i, 1, this.animateMsg(atbashEncode(el), true))
+        if(this.decrypted[i]) {
+          finalCiphertext.splice(i, 1, null); 
+          finalPlaintext.splice(i, 1, <div className="overlay is-inline-flex">{atbashEncode(el)}</div>)
+        }
+        else
+          this.decrypted[i] = true
+      } else {
+        finalCiphertext.splice(i, 1, <div className="overlay is-inline-flex">{el}</div>);
+        finalPlaintext.splice(i, 1, null);
+      }
+      console.log(finalCiphertext)
+    })
+
+    let transition;
+    if(this.state.animateDecrypt[0] && this.state.animateDecrypt[1] && this.state.animateDecrypt[2]) {
+      transition = <Anime opacity={[0,1]} translateX={[40, 0]} delay={anime.stagger(1000)}>
+        <br/>
+      <p class="subtitle">Not too shabby!</p>
+      <br />
+      <p class="subtitle">What's a <b>different</b> way you could transform one letter to another? </p>
+      <br/>
+      <p class="subtitle">Let's visit the Roman empire and see what kind of ciphers they've come up with...</p>
+    </Anime>
+    }
+    else { transition = <p class="is-size-5"></p> }
```

Comment:
You can simplify this down and boost performance if you swap out your array of state variables for a single one. I think you'd be surprised with what you can do with `map` as opposed to `splice`. We can discuss this one a little more in Slack since there's a lot to chew on here.

Reply:
Mind if I take the time to flesh that out this weekend?